OptimoveNotificationServiceExtension/Sources/OptimoveNotificationService.swift

```diff
@@ -1,217 +1,133 @@
-        let url = MediaHelper.getCompletePictureUrl(pictureUrl: picUrlNonNull as String, width: UInt(floor(UIScreen.main.bounds.size.width)))
-
-        dispatchGroup.enter()
-
-        loadAttachment(url!, withExtension: picExtension, completionHandler: { attachment in
-            if attachment != nil {
-                bestAttemptContent.attachments = [attachment!]
-            }
-            dispatchGroup.leave()
-        })
-    }
-
-    private class func getPictureExtension(_ pictureUrl: String?) -> String? {
-        if pictureUrl == nil {
-            return nil
+    static func buildIcon(button: PushNotification.Button) -> UNNotificationActionIcon? {
+        guard let icon = button.icon else { return nil }
+        switch icon.type {
+        case .custom:
+            return UNNotificationActionIcon(templateImageName: icon.id)
+        case .system:
+            return UNNotificationActionIcon(systemImageName: icon.id)
         }
-        let pictureExtension = URL(fileURLWithPath: pictureUrl!).pathExtension
-        if pictureExtension == "" {
-            return nil
-        }
-
-        return "." + pictureExtension
     }
 
-    private class func loadAttachment(_ url: URL, withExtension pictureExtension: String?, completionHandler: @escaping (UNNotificationAttachment?) -> Void) {
-        let session = URLSession(configuration: URLSessionConfiguration.default)
-
-        (session.downloadTask(with: url, completionHandler: { temporaryFileLocation, response, error in
-            if error != nil {
-                print("NotificationServiceExtension: \(error!.localizedDescription)")
-                completionHandler(nil)
-                return
-            }
-
-            var finalExt = pictureExtension
-            if finalExt == nil {
-                finalExt = self.getPictureExtension(response?.suggestedFilename)
-                if finalExt == nil {
-                    completionHandler(nil)
-                    return
-                }
-            }
-
-            if temporaryFileLocation == nil {
-                completionHandler(nil)
-                return
-            }
-
-            let fileManager = FileManager.default
-            let localURL = URL(fileURLWithPath: temporaryFileLocation!.path + finalExt!)
-            do {
-                try fileManager.moveItem(at: temporaryFileLocation!, to: localURL)
-            } catch {
-                completionHandler(nil)
-                return
-            }
-
-            var attachment: UNNotificationAttachment?
-            do {
-                attachment = try UNNotificationAttachment(identifier: "", url: localURL, options: nil)
-            } catch {
-                print("NotificationServiceExtension: attachment error: \(error.localizedDescription)")
-            }
-
-            completionHandler(attachment)
-        })).resume()
+    static func buildCategory(notification: PushNotification) async -> String {
```

Comment:
Worth renaming to register? can be registered in the caller instead. Just a suggestion